Source/WebCore/accessibility/AccessibilityRenderObject.cpp

```diff
@@ -2814,76 +2835,92 @@ void AccessibilityRenderObject::addChildren()
+            // AXChildIterator to walk the render tree / DOM (we may walk between the
+            // two — reference AccessibilityObject::iterator documentation for more information).
+            for (Ref object : AXChildIterator(*this))
+                addChildIfNeeded(WTFMove(object));
         }
-    } else {
-        if (auto* blockRenderer = dynamicDowncast<RenderBlock>(m_renderer.get()); blockRenderer && blockRenderer->isAnonymousBlock())
-            return;
-        // If we are a valid anonymous renderer (pseudo-element, list marker), use
-        // AXChildIterator to walk the render tree / DOM (we may walk between the
-        // two — reference AccessibilityObject::iterator documentation for more information).
-        for (Ref object : AXChildIterator(*this))
-            addChildIfNeeded(WTFMove(object));
-    }
 
-    if (RefPtr afterPseudo = element ? element->afterPseudoElement() : nullptr) {
-        if (RefPtr pseudoObject = cache ? cache->getOrCreate(*afterPseudo) : nullptr)
-            addChildIfNeeded(*pseudoObject);
-    }
+        if (RefPtr afterPseudo = element ? element->afterPseudoElement() : nullptr) {
+            if (RefPtr pseudoObject = cache ? cache->getOrCreate(*afterPseudo) : nullptr)
+                addChildIfNeeded(*pseudoObject);
+        }
 #else
-    // USE(ATPSI) within this block. Walk the render tree (primarily -- see comments for AccessibilityObject::iterator)
-    addListItemMarker();
-    // to build the accessibility tree.
-    // FIXME: Consider removing this ATSPI-only branch with https://bugs.webkit.org/show_bug.cgi?id=282117.
-    if (node && addListBoxChildrenIfNecessary(*node))
-        return;
+        // USE(ATPSI) within this block. Walk the render tree (primarily -- see comments for AccessibilityObject::iterator)
+        addListItemMarker();
+        // to build the accessibility tree.
+        // FIXME: Consider removing this ATSPI-only branch with https://bugs.webkit.org/show_bug.cgi?id=282117.
+        if (node && addListBoxChildrenIfNecessary(*node))
+            return;
 
-    for (auto& object : AXChildIterator(*this))
-        addChildIfNeeded(object);
+        for (auto& object : AXChildIterator(*this))
+            addChildIfNeeded(object);
 
-    addNodeOnlyChildren();
-    addCanvasChildren();
+        addNodeOnlyChildren();
+        addCanvasChildren();
 #endif // !USE(ATSPI)
 
-    addAttachmentChildren();
-    addImageMapChildren();
-    addTextFieldChildren();
-    addRemoteSVGChildren();
+        addAttachmentChildren();
+        addImageMapChildren();
+        addTextFieldChildren();
+        addRemoteSVGChildren();
 #if PLATFORM(MAC)
-    updateAttachmentViewParents();
+        updateAttachmentViewParents();
 #endif
-    updateOwnedChildren();
+        updateOwnedChildren();
+    }
+
+    // Handle aria-colindex for all table rows (whether using owned objects or not).
+    if (isExposedTableRow()) {
```

Comment:
We shouldn't call `isExposedTableRow()` twice in the span of one method call considering it does an ancestry traversal.